// @proofdict/textlint-rule-proofdict/lib/dictionary-storage.js
// MIT © 2018 azu
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var _require = require("localstorage-ponyfill"),
    createLocalStorage = _require.createLocalStorage;

var Storage = function () {
    function Storage() {
        _classCallCheck(this, Storage);

        this.localStorage = createLocalStorage();
    }

    _createClass(Storage, [{
        key: "getItem",
        value: function getItem(name, defaultValue) {
            return this.localStorage.getItem(name, defaultValue);
        }
    }, {
        key: "setItem",
        value: function setItem(name, value) {
            return this.localStorage.setItem(name, value);
        }
    }, {
        key: "removeItem",
        value: function removeItem(name) {
            return this.localStorage.removeItem(name);
        }
    }]);

    return Storage;
}();

var storage = exports.storage = new Storage();
